lib/chef/resource/windows_firewall_rule.rb

```diff
@@ -195,12 +236,38 @@ def firewall_command(cmdlet_type)
 
           cmd
         end
+
+        # return the current firewall rule settings for icmp_type and protocol are correct
+        # @return [Boolean]
+        def is_set_properly?(icmp_type, protocol)
+          if icmp_type.to_s.empty?
+            return false # rubocop:disable Style/RedundantReturn
+
+          elsif icmp_type.is_a?(Integer)
+            return false unless protocol.start_with?("ICMP")
```

Comment:
Do these checks need to be `protocol == "ICMP"`? Is a value of `ICMP2` valid?

Reply:
Yes, ICMPv4 and ICMPv6 are fine right now. This check is only needed for every protocol, which starts with ICMP, as every ICMP protocol requires an icmp_type. All other protocol types (which might become more) are only allowed to use icmp_type == 'Any'